Project1/PartyBase.h

```diff
@@ -0,0 +1,51 @@
+#pragma once
+#include "Unit.h"
+#include <list>
+
+class Map;
+class Party;
+class party_iterator;
+using UnitRef = std::reference_wrapper<Unit>;
+using constUnitRef = std::reference_wrapper<const Unit>;
+using UnitPtr = std::unique_ptr<Unit>;
+struct UnitData;
+struct PartyData {
+	std::string name;
+	std::vector<UnitData> unit_data;
+};
+class PartyBase {
+protected:
+	using iterator = std::list<UnitPtr>::iterator;
+	using const_iterator = std::list<UnitPtr>::const_iterator;
+
+	std::string _party_name;
+	std::list<UnitPtr> _units;
+
+	PartyBase();
+	PartyBase(std::string name);
+	PartyBase(std::string name, std::vector<UnitData> unit_data);
+	PartyBase(PartyData data);
+
```

Comment:
You need to explicitly declare the default destructor to make it protected.
`~PartyBase() = default;`